api/config/es_sidecar/config_request.go

```diff
@@ -141,4 +67,173 @@ func (c *ConfigRequest) SetGlobalConfig(g *ac.GlobalConfig) {
+
+	//
+	// Users can configure the ElasticSearch backup configuration the at es-sidecar
+	// level, the global backups level, and in some cases the global external
+	// level. Precedence for configuration is as follows in order of priority
+	//
+	// * External configuration (if using external ElasticSearch)
+	// * Global configuration
+	// * User defined es-sidecar configuration
+	// * es-sidecar default configuration
+	//
+	externalCfg := g.GetV1().GetExternal().GetElasticsearch()
+
+	backupsCfg := c.GetV1().GetSys().GetBackups()
+	if backupsCfg == nil {
+		backupsCfg = &ConfigRequest_V1_System_Backups{}
+	}
+
+	// If we're in external mode, apply and global and external config
+	if externalCfg.GetEnable().GetValue() {
+		c.applyGlobalBackupConfig(backupsCfg, g.GetV1().GetBackups())
+		c.applyExternalConfig(backupsCfg, externalCfg.GetBackup())
+		return
+	}
+
+	// If we're in internal mode, apply global config and then ensure that old
+	// internal fs config is migrated to use the backup gateway in S3 mode.
+	c.applyGlobalBackupConfig(backupsCfg, g.GetV1().GetBackups())
+	c.migrateToBackupGateway(backupsCfg)
+
+	return
+}
+
+// Given a backup config and an external configuration, merge any external values
+// into the backup config
+func (c *ConfigRequest) applyExternalConfig(
+	cfg *ConfigRequest_V1_System_Backups,
+	override *ac.External_Elasticsearch_Backup) {
+
+	// If backups are disabled we don't need to worry about applying any other
+	// config.
+	if !override.GetEnable().GetValue() {
+		cfg.Backend = w.String("disable")
+		return
+	}
+
+	backend := override.GetLocation().GetValue()
+	switch backend {
+	case "s3":
+		cfg.Backend = w.String(backend)
+		if cfg.S3 == nil {
+			cfg.S3 = &ConfigRequest_V1_System_Backups_S3Settings{}
+		}
+
+		extS3cfg := override.GetS3()
+		if b := extS3cfg.GetBucket(); b != nil {
+			cfg.S3.Bucket = b
+		}
+
+		if bp := extS3cfg.GetBasePath(); bp != nil {
+			cfg.S3.BasePath = bp
+		}
+
+		if c := extS3cfg.GetClient(); c != nil {
+			cfg.S3.Client = c
+		}
+
+		if s := extS3cfg.GetSettings(); s != nil {
+			cfg.S3.Es = s
+		}
+	case "fs":
+		cfg.Backend = w.String(backend)
+		if cfg.Fs == nil {
+			cfg.Fs = &ConfigRequest_V1_System_Backups_FsSettings{}
+		}
+
+		fsCfg := override.GetFs()
+		if p := fsCfg.GetPath(); p != nil {
+			cfg.Fs.RootLocation = p
+		}
+
+		if rb := fsCfg.GetSettings().GetMaxRestoreBytesPerSec(); rb != nil {
+			cfg.Fs.MaxRestoreBytesPerSec = rb
+		}
+
+		if sb := fsCfg.GetSettings().GetMaxSnapshotBytesPerSec(); sb != nil {
+			cfg.Fs.MaxSnapshotBytesPerSec = sb
+		}
+	default:
+		// Disable external backups by default
+		cfg.Backend = w.String("disabled")
+	}
+
+	return
+}
+
+// Given a backup config and the global backup config, merge any global values
+// into the backup config
+func (c *ConfigRequest) applyGlobalBackupConfig(
+	cfg *ConfigRequest_V1_System_Backups,
+	global *ac.Backups) {
+
+	if global == nil {
+		return
+	}
+
+	if backend := global.GetLocation(); backend != nil {
```

Comment:
should we be checking `GetValue()` here? would be things happen if it was an empty string?